drivers/wifi/esp32/wifi_esp32.c

```diff
@@ -0,0 +1,1879 @@
+/**
+ * Copyright (c) 2019 Manulytica Ltd
+ * Copyright (c) 2018 Linaro, Lmtd
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <logging/log.h>
+LOG_MODULE_REGISTER(wifi_esp32, CONFIG_WIFI_LOG_LEVEL);
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <zephyr.h>
+#include <kernel.h>
+#include <device.h>
+#include <gpio.h>
+#include <net/net_if.h>
+#include <net/net_offload.h>
+#include <net/net_pkt.h>
+#include <net/net_if.h>
+#include <net/net_context.h>
+#include <net/net_ip.h>
+#include <net/wifi_mgmt.h>
+
+#include "ipv4.h"
+#if defined(CONFIG_NET_UDP)
+#include "udp_internal.h"
+#endif
+
+#include "../../modem/modem_receiver.h"
```

Comment:
If you add `zephyr_library_include_directories` for drivers/modem to `CMakeLists.txt` above, then this can just be:
`#include "modem_receiver.h"`